packages/plugin-ext/src/plugin/languages/link-provider.ts

```diff
@@ -37,8 +48,9 @@ export class LinkProviderAdapter {
         }
 
         const doc = document.document;
-
         return Promise.resolve(this.provider.provideDocumentLinks(doc, token)).then(links => {
+            this.clearCache(doc.uri.toString());
```

Comment:
I've noticed that VS Code does not need a caches for tasks. Why do we need it? cc @RomanNikitenko could we just get rid of it?

An implementation in VS Code: https://github.com/Microsoft/vscode/blob/bd179feeb71a6195b6eab304452dfe7ac534e868/src/vs/workbench/api/node/extHostTask.ts#L582-L616

Reply:
I think we tested removing caches for tasks [here](https://github.com/theia-ide/theia/pull/5312/files#diff-bf710a40fc1eb5ef4a26ae139c63032eL24), but the PR was not merged (there are other causes of the problem, not related to cache for tasks).
So I believe - yes, we can get rid of caches for tasks.

Reply:
@RomanNikitenko it is good, i am going to dimiss approves there since you changed the pr and ping again for testing, could you rebase it?

Reply:
@akosyakov 
I answered [here](https://github.com/theia-ide/theia/pull/5312#issuecomment-520769303)